### PR TITLE
Staging/delay

### PR DESCRIPTION
**Pull request by danmois**
The activation of instruction cache for Xilinx projects. This is necessary for the correct operation of the usleep() and msleep() functions, that call usleep_MB(). The usleep_MB() function requires the enabling of the instruction cache on Xilinx platforms.

### CODE REVIEW

File: projects/ad9371/src/app/headless.c [hunk 1]
```diff
@@ -314,6 +314,12 @@ int main(void)
 
 	printf("Please wait...\n");
 
+	/* Instruction cache should be enabled for usleep functions to work. */
+	/* Enable the instruction cache. */
+	Xil_ICacheEnable();
```
Comment (buha) [inline projects/ad9371/src/app/headless.c:320]: this ad9371 change should probably go in the 2nd commit together with all the other such changes

**Comment by danmois** [comment]
V2

- reorganized commits so that they deal with the same modifications (1. delay functions. 2. activation of instruction/data cache)

**Review by dbogdan — APPROVED** [review]

**Review by adrimbarean — APPROVED** [review]
Looks good to me.

**Comment by dbogdan** [comment]
@danmois , I've just realized that it was not mentioned the main reason for enabling the Instruction Cache: it should be enabled for usleep_MB() to work.

**Comment by danmois** [comment]
V3

- the addition of text that describes the main reason for the modifications: Instruction Cache is enabled for the correct operation of usleep_MB().

**Pull request merged**